awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
+            (tz_now() - self.last_write).seconds > .1
+        )
+        if should_flush:
+            now = tz_now()
+            for cls, events in self.buff.items():
+                for e in events:
+                    e.created = e.modified = now
```

Comment:
I don't think this is what we want for `created` is it? What was the prior meaning of these two fields? I don't really know. It would be incredibly amazingly valuable to have a timestamp from the event callback. Someone suggested to me that this is what `created` is now, but I don't think that's right.

Reply:
My understanding of `.created` is that it was *always* something applied _at the time `.save()`_ was called, so I think this matches that behavior.